ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala

```diff
@@ -89,13 +92,14 @@ object KeyValueCommitting {
           logEntry -> outputs.toMap
 
         case DamlSubmission.PayloadCase.PARTY_ALLOCATION_ENTRY =>
-          ProcessPartyAllocation(
+          val (logEntry, outputs) = PartyAllocationCommitter.run(
             entryId,
             recordTime,
-            participantId,
             submission.getPartyAllocationEntry,
-            inputState
-          ).run
+            participantId,
+            inputState.collect { case (k, Some(v)) => k -> v }
```

Comment:
dropping the optional wrapping was a bad idea. we should keep it throughout to make sure that the submitter cannot cheat by "forgetting" to mention some inputs.